Lib/ast.py

```diff
@@ -346,7 +346,10 @@ def __instancecheck__(cls, inst):
             except AttributeError:
                 return False
             else:
-                return isinstance(value, _const_types[cls])
+                return (
+                    isinstance(value, _const_types[cls]) and
+                    not isinstance(value, _const_types_not[cls])
```

Comment:
```suggestion
                    not isinstance(value, _const_types_not.get(cls, ()))
```
and you can remove mappings to `()` in `_const_types_not`.